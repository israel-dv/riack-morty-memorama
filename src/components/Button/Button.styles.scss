@import '/src/styles/paletteColors';
@import '/src/styles/breakpoints';

%size-button {
  width: 240px;
  height: 53px;
  border-radius: 8px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: solid;
  border-width: 0 0 4px 4px;
  box-shadow: -1px 1px 4px rgba(0, 0, 0, 0.2);
  color: $text-primary;
}

.play {
  @extend %size-button;
  background-color: $blue-aqua;
  border-color: $green-lemon;
}
.start {
  background-color: $green-lemon;
  @extend %size-button;
  border-color: $blue-aqua;
}

.repeat {
  @extend %size-button;
  background-color: $blue-aqua;
  border-color: $green-lemon;
}

@media (max-width: $breakpoint-small) {
  %size-button {
    width: 140px;
  }
}

@media (max-width: $breakpoint-medium) {
  %size-button {
    width: 190px;
  }
}
