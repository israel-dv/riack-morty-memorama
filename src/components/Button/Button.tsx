import { Typography } from 'components/Typography'

import './Button.styles.scss'

type ButtonProps = {
  type?: 'play' | 'repeat' | 'start'
  onClick?: () => void
}

const BUTTON_TITLE = {
  play: 'Jugar',
  repeat: 'Repetir',
  start: 'Iniciar',
}

export const Button = ({ type = 'play', onClick }: ButtonProps): JSX.Element => {
  return (
    <button className={`btn-game ${type}`} onClick={onClick}>
      <Typography.Heading2 text={BUTTON_TITLE[type]} />
    </button>
  )
}
