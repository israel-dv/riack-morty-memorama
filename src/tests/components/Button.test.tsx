import { render } from '@testing-library/react'

import { Button } from '../../components/Button'

describe('Test Button to show just 3 options', () => {
  test('Repeat Button', async () => {
    const { container } = render(<Button type="repeat" />)
    expect(container).toMatchSnapshot()
  })
  test('Start Button', async () => {
    const { container } = render(<Button type="start" />)
    expect(container).toMatchSnapshot()
  })
  test('Play Button', async () => {
    const { container } = render(<Button type="play" />)
    expect(container).toMatchSnapshot()
  })
})
