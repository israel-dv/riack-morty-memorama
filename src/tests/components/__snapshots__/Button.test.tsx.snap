// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button to show just 3 options Play Button 1`] = `
<div>
  <button
    class="btn-game play"
  >
    <h2
      class="heading2"
    >
      Jugar
    </h2>
  </button>
</div>
`;

exports[`Test Button to show just 3 options Repeat Button 1`] = `
<div>
  <button
    class="btn-game repeat"
  >
    <h2
      class="heading2"
    >
      Repetir
    </h2>
  </button>
</div>
`;

exports[`Test Button to show just 3 options Start Button 1`] = `
<div>
  <button
    class="btn-game start"
  >
    <h2
      class="heading2"
    >
      Iniciar
    </h2>
  </button>
</div>
`;
